applications/FluidDynamicsHydraulicsApplication/custom_utilities/hydraulic_fluid_auxiliary_utilities.cpp

```diff
@@ -0,0 +1,322 @@
+    }
+    else
+    {
+        distance_getter = [](NodeType &rNode, const Variable<double> &rDistanceVariable) -> double
+        { return rNode.GetValue(rDistanceVariable); };
+    }
+
+    KRATOS_ERROR_IF(rModelPart.GetProcessInfo()[DOMAIN_SIZE] < 3) << "Wetted perimeter is only implemented for 3D." << std::endl;
+
+    // Auxiliary container for the fake Triangle2D3 geometries' points
+    GeometryType::PointsArrayType aux_points;
+    double wetted_area = 0.0;
+
+    for (auto &r_cond : rModelPart.Conditions())
+    {   // Create an auxiliary element based on the rskinflag condition.
+        if (r_cond.Is(rSkinFlag))
+        {
+            std::vector<ModelPart::IndexType> elem_nodes_id;
+            auto &r_geom = r_cond.GetGeometry();
+
+            // Fill the points array and distances vector
+            Vector aux_distances(3);
+            for (IndexType i_nodes = 0; i_nodes < r_geom.PointsNumber(); i_nodes++)
+            {
+                aux_points.push_back(r_geom(i_nodes));
+                aux_distances[i_nodes] = r_geom[i_nodes].GetValue(rDistanceVariable);
+                // TODO: It should be posible to have an historical distance variable.
+                // aux_distances[i_nodes] =distance_getter;
+            }
+            // Calculate the water area (wetted area) of the cut conditions
+            if (IsSplit(aux_distances))
+            {
+                auto p_splitting_util = Kratos::make_unique<DivideTriangle3D3<NodeType>>(r_geom, aux_distances);
+                p_splitting_util->GenerateDivision();
+                const auto& r_neg_subdivisions = p_splitting_util->GetNegativeSubdivisions();
+                for (const auto& rp_neg_subdivision : r_neg_subdivisions) {
+                    wetted_area += rp_neg_subdivision->Area();
+                }
+            }
+            // Calculate the water area (wetted area)
+            else if (IsNegative(aux_distances))
+            {
+                wetted_area += r_geom.Area();
+            }
+            aux_points.clear();
+        }
+    }
+    return wetted_area;
+}
+
+bool HydraulicFluidAuxiliaryUtilities::IsSplit(const Vector &rConditionDistancesVector)
+{
+    std::size_t n_pos(0);
+    std::size_t n_neg(0);
+    const std::size_t pts_number = rConditionDistancesVector.size();
+    for (std::size_t i_node = 0; i_node < pts_number; ++i_node)
+    {
+        if (rConditionDistancesVector[i_node] > 0.0)
+            n_pos++;
+        else
+            n_neg++;
+    }
+    return (n_pos > 0 && n_neg > 0) ? true : false;
+}
+
+bool HydraulicFluidAuxiliaryUtilities::IsNegative(const Vector &rConditionDistancesVector)
+{
+    std::size_t n_neg(0);
+    const std::size_t pts_number = rConditionDistancesVector.size();
+    for (std::size_t i_node = 0; i_node < pts_number; ++i_node)
+    {
+        if (rConditionDistancesVector[i_node] < 0.0)
+            n_neg++;
+    }
+    return n_neg == pts_number;
+}
```

Comment:
Aren't these the very same implementation we have in the `FluidDynamicsApplication` utilities? If so I'd include those in here instead of copying and pasting (note that you can perfectly do it as your app has compile dependency to the standard fluid app, i.e., the library is linked to the `FluidDynamicsApplication`'s one).

Reply:
Yes, you are right